tests/unit/commands/test_push.py

```diff
@@ -152,10 +152,41 @@ def json(self):
                     ]
                 )
 
-        self.fake_store_push_precheck.mock.side_effect = StorePushError(
-            "basic", MockResponse()
+        self.fake_store_push_precheck.mock.side_effect = [
+            StorePushError("basic", MockResponse()),
+            None,
+        ]
+
+        result = self.run_command(["push", self.snap_file], input="y\n")
+
+        self.assertThat(result.exit_code, Equals(0))
```

Comment:
out of curiosity, why not use the short form, self.assertEquals(reference, actual) ?  I tend to use the short-form versions, but notice you use assertThat instead...

```
self.assertEquals(0, result.exit_code)
self.assertRegex(result.output, "You are required to register this snap before continuing.")
```
seems simpler to me... but maybe you prefer the sentence-like structure? :D

Reply:
Consistency, this is probably another thing we should add to the style guide, something agreed as a team way back...